components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -432,24 +472,56 @@ protected String getLoginType(HttpServletRequest request) {
         }
     }
 
-    @Override
-    public String getClaimDialectURI() {
-        String claimDialectUri = null;
+    private Map<String, String> readParametersFromAuthenticatorConfig() {
         AuthenticatorConfig authConfig = FileBasedConfigurationBuilder.getInstance().getAuthenticatorBean(getName());
         if (authConfig != null) {
-            Map<String, String> parameters = authConfig.getParameterMap();
-            if (parameters != null && parameters.containsKey(FacebookAuthenticatorConstants.
-                    CLAIM_DIALECT_URI_PARAMETER)) {
-                claimDialectUri = parameters.get(FacebookAuthenticatorConstants.CLAIM_DIALECT_URI_PARAMETER);
+            return authConfig.getParameterMap();
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("FileBasedConfigBuilder returned null AuthenticatorConfigs for the connector " +
+                        getName());
+            }
+            return null;
```

Comment:
Never return a null if the return type is a map. Instead you can use collections. empty map